File: backend/src/main/java/com/example/backend/service/ProductService.java
package com.example.backend.service;

import com.example.backend.entity.Product;
import com.example.backend.repository.ProductRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
public class ProductService {

    @Autowired
    private ProductRepository productRepository;

    // Método para crear un producto
    public Product createProduct(Product product) {
        return productRepository.save(product);
    }

    // Método para obtener todos los productos
    public List<Product> findAllProducts() {
        return productRepository.findAll();
    }

    // Método para obtener un producto por ID
    public Optional<Product> findProductById(Long id) {
        return productRepository.findById(id);
    }

    // Método para actualizar un producto existente
    public Product updateProduct(Long id, Product updatedProduct) {
        return productRepository.findById(id)
            .map(existingProduct -> {
                existingProduct.setNombre(updatedProduct.getNombre());
                existingProduct.setCategoria(updatedProduct.getCategoria());
                existingProduct.setPrecio(updatedProduct.getPrecio());
                existingProduct.setCantidad(updatedProduct.getCantidad());
                return productRepository.save(existingProduct);
            })
            .orElseThrow(() -> new RuntimeException("Producto no encontrado"));
    }

    // Método para eliminar un producto
    public void deleteProduct(Long id) {
        productRepository.deleteById(id);
    }
}

File: frontend/src/services/productService.js
import axios from "axios";

const API_URL = "http://localhost:8080/products";

// Crear un producto
export const createProduct = async (product) => {
  const response = await axios.post(API_URL, product);
  return response.data;
};

// Actualizar un producto
export const updateProduct = async (id, product) => {
  const response = await axios.put(`${API_URL}/${id}`, product);
  return response.data;
};

// Eliminar un producto
export const deleteProduct = async (id) => {
  const response = await axios.delete(`${API_URL}/${id}`);
  return response.data;
};

// Obtener todos los productos
export const fetchProducts = async () => {
  const response = await axios.get(API_URL);
  return response.data;
};
